test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,326 @@
+import inspect
+import re
+from typing import get_type_hints
+from unittest import mock
+
+import PIL.Image
+import pytest
+
+import torch
+from common_utils import cache, cpu_and_gpu, make_bounding_box, make_image
+from torch.testing import assert_close
+from torch.utils._pytree import tree_map
+from torchvision import datapoints
+from torchvision.transforms.v2 import functional as F
+from torchvision.transforms.v2.utils import is_simple_tensor
+
+
+def _to_tolerances(maybe_tolerance_dict):
+    default_tolerances = dict(rtol=None, atol=None)
+    if not isinstance(maybe_tolerance_dict, dict):
+        return default_tolerances
+
+    missing = default_tolerances.keys() - maybe_tolerance_dict.keys()
+    if missing:
+        raise pytest.UsageError("ADDME")
+
+    extra = maybe_tolerance_dict.keys() - default_tolerances.keys()
+    if extra:
+        raise pytest.UsageError("ADDME")
+
+    return maybe_tolerance_dict
```

Comment:
> The whole logic looks overkill, we don't really need all those checks. If we pass `aatol` instead of `atol`, we're going to know soon enough.

I'm ok with relying on the default errors that will be coming from `torch.testing.assert_close`. Note however, that your suggestion

```python
maybe_tolerance_dict = maybe_tolerance_dict if isinstance(maybe_tolerance_dict, dict) else {}
return dict(rtol=None, atol=None, **maybe_tolerance_dict)
```

either is also too complicated or doesn't work. `assert_close` does not accept only `rtol` or `atol`. So either we need to set the default to `0` (a) or we implicitly require to always set both (b).

(a)

```python
if not isinstance(maybe_tolerance_dict, dict):
    return dict(rtol=None, atol=None)
return return dict(rtol=0, atol=0, **maybe_tolerance_dict)
```

(b)

```python
if isinstance(maybe_tolerance_dict, dict):
    # if not both rtol and atol are supplied, this will fail later on
    return maybe_tolerance_dict
return dict(rtol=None, atol=None)
```

> We can also pass `atol` and `rtol` as-is to the `_check_xyz` functions instead of packing them in dicts?

We can, yes.

Reply:
I don't understand. There's a problem with my code above and it should be `    return {"rtol": None, "atol": None, **maybe_tolerance_dict}` instead of `dict(...)`

But other than that, everything works as expected. If one passes `check_scripted_vs_eager={"atol": 1}` we get

```
ValueError: Both 'rtol' and 'atol' must be either specified or omitted, but got no rtol.
```

That LGTM?

Reply:
You are right, I misunderstood your implementation. Given

> That's an undesirable limitation of `assert_close` IMHO

do you still want to go with this as default or should we go with (a) above? The latter would mean `check_scripted_vs_eager={"atol": 1}` would be mapped to `{"atol": 0, "rtol": 0}`.

I agree BTW on the `assert_close` limitation. IMO it would have been better to just map an omitted parameter to `0` unless both are omitted in which case, the default tolerances can be used.

Reply:
No strong preference, a) looks fine